### ISSUE TITLE
Inventory creation/deletion/double reservation problem

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1467]
_Imported from trac ticket 1467.
Created by **mariam**
Opened in trac: Feb 21, 2012, 9:59:14 AM
Last modified in trac: Feb 27, 2012, 3:23:03 PM_
***
_by mariam:_
See also [#1469](1469)

Location: 17/STORE

Order: 0127217KC93
SKU: 716114258

This has happened a few times before but they have been fixing them.

Originally we had 2 at this location.

One of them was sold to this customer but for some reason the same order seemed to have reserved 2 pieces at almost the same time with a few seconds difference.

There was not unreserved transaction in between.

The Inventory already adjusted the inventory +1 at location 17.

These are the transactions in ITQ for this item :
```
DEL_DOC_NUM	QTY	TRN_DT	 TIME	        ITT
0127217KC93	1	01/27/12 05:53:34 PM	SOF
0127217KC93	1	01/27/12 04:33:11 PM	OOR
0127217KC93	1	01/27/12 04:33:23 PM	OOR
```Mariam H. Farach

**Comment by kbower** [comment]
_**kbower** trac comment on Feb 21, 2012, 10:17:32 AM:_

Another example would be very good. To make sure I understand: in the end, is fictitious inventory being created?

**Comment by kbower** [comment]
_**kbower** trac comment on Feb 21, 2012, 10:21:41 AM:_

-------- Original Message --------
Subject:         RE: Re #1467: Inventory Issues
Date:         Tue, 21 Feb 2012 10:22:03 -0500
From:         Mariam Farach <Mariam@ElDoradoFurniture.com>
To:         Kent Bower <kb@retailarchitects.com>



Yes.

**Comment by kbower** [comment]
_**kbower** trac comment on Feb 21, 2012, 3:35:30 PM:_

```
SQL> select CREATEDATE, SOURCE, TYPE, PRODUCTID, QTY, OLDSITEID, OLDLOCATIONID, NEWSITEID, NEWLOCATIONID from inventorytrans where orderid = '0127217KC93' order by 1
  2  ;

CREATEDATE           SOURC TYP PRODUCTID        QTY OL OLDLOC NE NEWLOC
-------------------- ----- --- --------- ---------- -- ------ -- ------
27-JAN-2012 16:33:11 SOM   OOR 716114258          1 17 STORE
27-JAN-2012 16:33:23 SOM   OOR 716114258          1 17 STORE
27-JAN-2012 17:53:34 SOM   SOF 716114258          1 17 STORE

```

**Comment by kbower** [comment]
_**kbower** trac comment on Feb 23, 2012, 7:47:42 PM:_

(Changeset [[2311]](../commit/7e5d2a5a201d97de3a9e6d8d322c9cba301104b0))
 * Fix concurrent change problem so we can detect if another process (e.g. GERS) has changed the data we are also editing.
   * Fixes #1467 and [#1469](1469) (Inventory creation/deletion/reservation/unreservation problems occuring when Order is edited in GERS and open in Loft)

**Comment by kbower** [comment]
_**kbower** trac status change on Feb 23, 2012, 7:47:42 PM:_ **closed**
